id/id.go

```diff
@@ -0,0 +1,20 @@
+// Package id contains facilities for generating and validating random IDs
+// which are meant to be used as uploaded filenames.
+package id
+
+import (
+	"crypto/rand"
+	"math"
+	"math/big"
+)
+
+// Generate a random ID like "1njfizqgeukrq", "3l44ze7pf47fd", "35dgd4n5ryup", etc.
+func Generate() (id string, err error) {
```

Comment:
I would prefer a function that takes the output string length and a collection of possible characters to put into the output string, but this will do.